### PR TITLE
form Dotácia kontajnerové stanovištia, support allOf and if-then

### PR DESCRIPTION
**Pull request by kristianduda**

### CODE REVIEW

File: __tests__/utils/validation.test.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { cwd } from 'node:process'
 import { ErrorType, loadAndValidate } from '../../src/utils/validation'
 
 describe('validation', () => {
+  const bodyElement = `xs\\:element[name='Body']`;
```
Comment (mpinter) [inline __tests__/utils/validation.test.ts:8]: besides this PR, but I remember having hard time finding a reasonable source of docs for writing these path expressions (this is "xpath", right ? or something else ?) - if you are using a certain source, or a collection of sources, can you please link them in README or in the docs ?

File: src/utils/forms.ts [hunk 2]
```diff
@@ -71,6 +73,19 @@ const getJsonSchemaFormat = (type: string | undefined): JsonSchemaFormat => {
...
+}
+
+export const getJsonSchemaProperties = (jsonSchema: JsonSchema): JsonSchemaProperties => {
+  let properties: JsonSchemaProperties = jsonSchema.properties ?? {}
+  if (jsonSchema.then) {
+    properties = { ...properties, ...jsonSchema.then.properties }
+  }
+
+  return properties
+}
```
Comment (mpinter) [inline src/utils/forms.ts]: would like this to somehow imply the fact this is not just unwrapping of .properties, but you also get all of the potential properties (and who knows what in the future if we start dealing with `oneOf`)

what about `getAllPossibleJsonSchemaProperties` ?

File: package.json [hunk 3]
```diff
@@ -57,6 +57,7 @@
   },
   "dependencies": {
     "chalk": "2.4",
+    "json-schema-merge-allof": "^0.8.1",
```
Comment (mpinter) [inline package.json:60]: noticing just now, but don't we technically need `lodash` and `cheerio` as dependencies, instead of devDependencies ?

**Review by mpinter — APPROVED** [review]
✅

File: __tests__/utils/validation.test.ts (hunk 1 above)
Comment (kristianduda) [reply __tests__/utils/validation.test.ts:8]: I'm not using any specific docs, as I have worked with xpath before. If I find something, will add it to the readme.

File: package.json (hunk 3 above)
Comment (kristianduda) [reply package.json:60]: Don't think so, as `lodash` and `cheerio` are peer dependencies.

**Pull request merged**